docs/rechunking.ipynb

```diff
@@ -0,0 +1,418 @@
+      "WARNING:root:Make sure that locally built Python SDK docker image has Python 3.9 interpreter.\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "[(Key(offsets={'x': 0, 'y': 0}, vars={'foo'}),\n",
+       "  <xarray.Dataset>\n",
+       "  Dimensions:  (x: 4, y: 2)\n",
+       "  Dimensions without coordinates: x, y\n",
+       "  Data variables:\n",
+       "      foo      (x, y) int64 0 1 2 3 4 5 6 7),\n",
+       " (Key(offsets={'x': 0, 'y': 0}, vars={'bar'}),\n",
+       "  <xarray.Dataset>\n",
+       "  Dimensions:  (x: 4, y: 2)\n",
+       "  Dimensions without coordinates: x, y\n",
+       "  Data variables:\n",
+       "      bar      (x, y) int64 100 101 102 103 104 105 106 107),\n",
+       " (Key(offsets={'x': 4, 'y': 0}, vars={'foo'}),\n",
+       "  <xarray.Dataset>\n",
+       "  Dimensions:  (x: 4, y: 2)\n",
+       "  Dimensions without coordinates: x, y\n",
+       "  Data variables:\n",
+       "      foo      (x, y) int64 8 9 10 11 12 13 14 15),\n",
+       " (Key(offsets={'x': 4, 'y': 0}, vars={'bar'}),\n",
+       "  <xarray.Dataset>\n",
+       "  Dimensions:  (x: 4, y: 2)\n",
+       "  Dimensions without coordinates: x, y\n",
+       "  Data variables:\n",
+       "      bar      (x, y) int64 108 109 110 111 112 113 114 115)]"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "inputs | xbeam.SplitVariables()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Adjusting chunks"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also adjust _chunks_ in a dataset to distribute arrays of different sizes. Here you have two choices of API:\n",
+    "\n",
+    "1. The lower level {py:class}`~xarray_beam.SplitChunks` and {py:class}`~xarray_beam.ConsolidateChunks`. These transformations apply a single splitting (with indexing) or consolidation (with {py:function}`xarray.concat`) function to array elements.\n",
+    "2. The high level {py:class}`~xarray_beam.Rechunk`, which uses a pipeline of multiple split/consolidate steps to efficient rechunk a dataset."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For minor adjustments, the more explicit `SplitChunks()` and `ConsolidateChunks()` are good options. They take a dict of _desired_ chunk sizes as a parameter, which can also be `-1` to indicate \"no chunking\" along a dimension:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:apache_beam.options.pipeline_options:Discarding unparseable args: ['/Users/shoyer/miniconda3/envs/xarray-beam/lib/python3.9/site-packages/ipykernel_launcher.py', '-f', '/Users/shoyer/Library/Jupyter/runtime/kernel-2b08a4f7-b064-490e-9ac1-e22ceca4c6cd.json']\n",
```

Comment:
An idea: Maybe, modify the JSON of the notebook to get rid of these warnings?

Reply:
I _could_ do this, but I don't think it's worth the hassle of manual edits every time I change a notebook.

Reply:
One options: We could use nbconvert to clean the output of notebooks in CI. Though, this may be throwing the baby out with the bathwater. 

https://stackoverflow.com/a/47774393